src/dsl/cb.rs

```diff
@@ -167,9 +239,15 @@ pub fn when<F: From<u64> + Clone, T1: Into<Constraint<F>>, T2: Into<Constraint<F
     let selector = selector.into();
     let when_true = when_true.into();
 
+    match selector.typing {
+        Typing::Boolean => (),
+        _ => panic!("Expected boolean selector, got {:?}, selector: {}", selector.typing, selector.annotation)
+    }
+
     Constraint {
         annotation: format!("if({})then({})", selector.annotation, when_true.annotation),
         expr: selector.expr * when_true.expr,
+        typing: when_true.typing, // bytecode example used `when` as an AntiBooly, but I think we can also use it as a Boolean
```

Comment:
Comment not clear and I think it is not necessary

Reply:
Deleted